frontend/app/components/wp-panels/watchers-panel/watchers-panel.controller.ts

```diff
@@ -28,120 +28,107 @@
 
 import {WorkPackageResourceInterface} from '../../api/api-v3/hal-resources/work-package-resource.service';
 import {WorkPackageCacheService} from '../../work-packages/work-package-cache.service';
+import {WorkPackageNotificationService} from '../../wp-edit/wp-notification.service';
+import {CollectionResource} from '../../api/api-v3/hal-resources/collection-resource.service';
 
 export class WatchersPanelController {
 
-  public workPackage:WorkPackageResourceInterface;
-  public loadingPromise:ng.IPromise<any>;
+  public workPackage: WorkPackageResourceInterface;
+  public loadingPromise: ng.IPromise<any>;
+  public autocompleteLoadingPromise: ng.IPromise<any>;
+  public autocompleteInput = '';
   public error = false;
   public allowedToAdd = false;
   public allowedToRemove = false;
 
-  public available:any[] = [];
-  public watching:any[] = [];
-  public text:any;
+  public watching: any[] = [];
+  public text: any;
 
   constructor(public $scope,
+              public $q,
               public I18n,
-              public wpCacheService:WorkPackageCacheService,
-              public wpWatchers) {
+              public wpNotificationsService: WorkPackageNotificationService,
+              public wpCacheService: WorkPackageCacheService) {
 
     this.text = {
       loading: I18n.t('js.watchers.label_loading'),
-      loadingError: I18n.t('js.watchers.label_error_loading')
+      loadingError: I18n.t('js.watchers.label_error_loading'),
+      autocomplete: {
+        placeholder: I18n.t('js.watchers.label_click_to_add')
```

Comment:
We should have a different label here:
'Click to add watchers' is not correct, as I need to start typing before options are presented. And personally, I was somewhat confused regarding how I would have to use the widget and I have already used it.

If easy, we could also change the behaviour to issue a first request without a filter once the user clicks into the field to already present a selection to the user. Oftentimes, as most projects will not have more than thirty members, this will already suffice to select the desired user.